integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -186,54 +161,51 @@ spec:
           - containerPort: 20002
             name: job-web
           volumeMounts:
-          {{- if $root.Values.secrets }}
-            {{- if $root.Values.secrets.master }}
-              {{- range $key, $val := $root.Values.secrets.master }}
-            - name: secret-{{ $key }}-volume
-              mountPath: /secrets/{{ $val }}
-              readOnly: true
-              {{- end }}
+          {{- if .Values.secrets }}
+            {{- if .Values.secrets.master }}
+{{- include "alluxio.master.secretVolumeMounts" . | indent 10 }}
             {{- end }}
           {{- end }}
-          {{- if $root.Values.mounts }}
-            {{- range $root.Values.mounts }}
+          {{- if .Values.mounts }}
+            {{- range .Values.mounts }}
             - name: "{{ .name }}"
               mountPath: "{{ .path }}"
             {{- end }}
           {{- end }}
       restartPolicy: Always
       volumes:
-        {{- if $isSingleUfsLocal }}
-        - name: alluxio-journal
-          persistentVolumeClaim:
-            claimName: {{ $root.Values.journal.pvcName }}
-        {{- end }}
-        {{- if $root.Values.master.metastore }}
+        {{- if .Values.master.metastore }}
         - name: alluxio-metastore
           emptyDir:
-            medium: {{ $root.Values.master.metastore.medium }}
-            sizeLimit: {{ $root.Values.master.metastore.size | quote }}
+            medium: {{ .Values.master.metastore.medium }}
+            sizeLimit: {{ .Values.master.metastore.size | quote }}
         {{- end }}
-        {{- if $isHaEmbedded }}
-        - name: alluxio-journal
-          emptyDir: {}
-        {{- end}}
-        {{- if $root.Values.secrets }}
-          {{- if $root.Values.secrets.master }}
-            {{- range $key, $val := $root.Values.secrets.master }}
+        {{- if .Values.secrets }}
+          {{- if .Values.secrets.master }}
+            {{- range $key, $val := .Values.secrets.master }}
         - name: secret-{{ $key }}-volume
           secret:
             secretName: {{ $key }}
             defaultMode: 256
             {{- end }}
           {{- end }}
         {{- end }}
-        {{- if $root.Values.mounts }}
-          {{- range $root.Values.mounts }}
+        {{- if .Values.mounts }}
+          {{- range .Values.mounts }}
         - name: "{{ .name }}"
           persistentVolumeClaim:
             claimName: "{{ .name }}"
           {{- end }}
         {{- end }}
----
-{{- end }}
+  {{- if $needJournalVolume }}
+  volumeClaimTemplates:
+    - metadata:
+        name: alluxio-journal
+      spec:
+        storageClassName: standard
+        accessModes:
+          - ReadWriteMany
+        resources:
+          requests:
+            storage: {{ .Values.journal.size }}
+  {{- end }}
```

Comment:
Use volumeClaimTemplates now instead of manually defining PVC for journal. This will create a PVC for each Pod in the StatefulSet, with a deterministic name.